src/storage/src/table/mem_table.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#![allow(dead_code)]
+#![allow(unused)]
+use std::collections::HashMap;
+
+use risingwave_common::array::Row;
+
+use crate::error::StorageResult;
+
+pub enum RowOp {
+    Insert(Row),
+    Delete(Row),
+    Update((Row, Row)),
+}
+pub struct MemTable {
+    pub buffer: HashMap<Row, RowOp>,
+}
+impl Default for MemTable {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+impl MemTable {
```

Comment:
A `MemTable::drain(&mut self)` function returns a Iterator of `(pk: Row, op: RowOp)` and clear the memtable.
It will be called by `StateTable::commit` and drain all the operators in `mem_table` and flush them into `cell_based_table`